pkg/build/api/validation/validation_test.go

```diff
@@ -505,3 +505,59 @@ func TestValidateTrigger(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateImageChange(t *testing.T) {
+	tests := []struct {
+		name           string
+		ict            *buildapi.ImageChangeTrigger
+		expectedErrNum int
+	}{
+		{
+			name: "Pass",
+			ict: &buildapi.ImageChangeTrigger{
+				Image: "openshift",
+				From: kapi.ObjectReference{
+					Name: "default/repo",
+				},
+			},
+			expectedErrNum: 0,
+		},
+		{
+			name: "Missing image ref",
+			ict: &buildapi.ImageChangeTrigger{
+				Image: "",
+				From: kapi.ObjectReference{
+					Name: "default/repo",
+				},
+			},
+			expectedErrNum: 1,
+		},
+		{
+			name: "Missing From ref",
+			ict: &buildapi.ImageChangeTrigger{
+				Image: "openshift",
+				From: kapi.ObjectReference{
+					Name: "",
+				},
+			},
+			expectedErrNum: 1,
+		},
+		{
+			name: "Both missing refs",
+			ict: &buildapi.ImageChangeTrigger{
+				Image: "",
+				From: kapi.ObjectReference{
```

Comment:
how about a test where you don't even define the From field?  Just in case someone changes it to a ptr in the future and sets us up for a nil ptr error.